### PR TITLE
Follow-up polishing of creative

### PR DESCRIPTION
**Pull request by kilbith**
Some basic maintenance to conclude the refresh of `creative` : 
- Merge two conditions into one.
- Simplify a Lua pattern.
- New (small) feature : go to the last page on clicking on [<<] at page 1.

### CODE REVIEW

**Comment by paramat** [comment]
```
	if not (def.groups.not_in_creative_inventory == 1) and
				def.description and def.description ~= ""
```
Is simple and fast while:
```
if (filter and def.name:find(filter, 1, true)) or not filter
```
.. looks to be a slower operation, so having 2 'if' statements might actually be the optimum.

Making the inventory pages wrap is a good feature i have been wanting.

**Comment by kilbith** [comment]
Please stop these pointless comments on the `if` statements.

I've done some benchmarks and got an equal average of speed on this piece of code : 

``` python
Before (initial) : 0.65 ms
Before (filtering) : 1.13 ms

After (initial) : 0.34 ms
After (filtering) : 0.75 ms
```

It's processed the same way than before with no impact on speed.

File: mods/creative/init.lua [hunk 1]
```diff
@@ -51,10 +51,9 @@ function creative_inventory.update(player_name, filter)
 
 	for name, def in pairs(minetest.registered_items) do
 		if not (def.groups.not_in_creative_inventory == 1) and
-				def.description and def.description ~= "" then
-			if (filter and def.name:find(filter, 1, true)) or not filter then
-				creative_list[#creative_list+1] = name
-			end
+				def.description and def.description ~= "" and
+				((filter and def.name:find(filter, 1, true)) or not filter) then
```
Comment (rubenwardy) [inline mods/creative/init.lua]:
```lua
((filter and def.name:find(filter, 1, true)) or not filter)
```

would be better as

```lua
not filter or def.name:find(filter, 1, true)
```

**Comment by rubenwardy** [comment]
> .. looks to be a slower operation, so having 2 'if' statements might actually be the optimum.

If you have an if statement like so:

```lua
if false and foo and bar then
```

Then foo and bar will never be executed. It's like this in pretty much every language.

**Comment by paramat** [comment]
Thanks rubenwardy i see.
The question was valid not pointless.

**Comment by paramat** [comment]
:+1:

**Comment by Ekdohibs** [comment]
Looks good!

**Comment by paramat** [comment]
Will merge soon.

**Comment by paramat** [comment]
7488f483c70ce1c9f804259b144cbfc475b9bf1d

**Pull request closed without merging**